js/main.js

```diff
@@ -35,12 +44,12 @@ function toList(collection) {
 function calculateTotalMats(elementAmount) {
     let collection = {};
     for (let key of Object.keys(crafting)) { // key = key of crafting Object
-        for (let value of crafting[key]) { // value = value associated with key from crafting Object
+        for (let value of crafting[key].materials) { // value = value associated with key from crafting Object
             let userInput = Number(elementAmount[key]); // userInput = each value associated with the key in elementAmount
             if (!Number.isInteger(userInput) || userInput <= 0) {
                 continue;
             }
-            if (!collection.hasOwnProperty(userInput)) {
+            if (!collection.hasOwnProperty(value.material)) {

```

Comment:
You now use `value.material` 3 times. Might be worth a variable at this point

Reply:
Fixed